src/debugger/godot4/server_controller.ts

```diff
@@ -411,8 +416,12 @@ export class ServerController {
 				break;
 			}
 			case "stack_frame_vars": {
-				this.partialStackVars.reset(command.parameters[0]);
-				this.session.set_scopes(this.partialStackVars);
+				const stack_var_count = command.parameters[0];
+				this.partialStackVars.reset(stack_var_count);
+
+				if(stack_var_count <= 0)
+					this.session.set_scopes(this.partialStackVars);
```

Comment:
if set_scopes is called here every time the stack_frame_vars command is handled, it leads to the issue seen in the second image in the following comment:

> Restarting debugging, with custom_object expanded from previous debug session. ![image](https://private-user-images.githubusercontent.com/61250140/337883845-d5743994-0cad-46a2-992c-ce4e3a7aeed3.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTc4NTk1MjEsIm5iZiI6MTcxNzg1OTIyMSwicGF0aCI6Ii82MTI1MDE0MC8zMzc4ODM4NDUtZDU3NDM5OTQtMGNhZC00NmEyLTk5MmMtY2U0ZTNhN2FlZWQzLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDA2MDglMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwNjA4VDE1MDcwMVomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWJiOTllNjEzZDc3ZTFhZmEyYTQzZjI5ODY4NTM3ODU0ZTNiYmQ3M2Y0NWMxMzg3YzZhYWZiMGJiMmQ1NDA1YjAmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.wNpCgNxbHZiG_BKMppEHcs2P1e8Ob3HONW7aXKzFI-c)

https://github.com/godotengine/godot-vscode-plugin/pull/662#issuecomment-2156068748


This if statement makes sure GodotDebugSession.ongoing_inspections is not of length 0 (if Objects are present in the current stack frame) the first time set_scopes is called, thus preventing GodotDebugSession.got_scope.notify being called too early.